servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp

```diff
@@ -1379,7 +1379,7 @@ void RendererCanvasRenderRD::canvas_render_items(RID p_to_render_target, Item *p
 	while (ci) {
 		if (ci->copy_back_buffer && canvas_group_owner == nullptr) {
 			backbuffer_copy = true;
-
+			backbuffer_gen_mipmaps = true;
```

Comment:
The backbuffer should only generate mipmaps if they will actually be used.

Reply:
With a CanvasItem that has a screen-reading shader you can determine whether mipmaps should be generated (via `shader_data->uses_screen_texture_mipmaps`). I might be wrong, but the problem in this case is that with a `BackBufferCopy` node you can't know if a CanvasItem uses the mipmaps later with our current canvas renderer implementation. If we want control over that we would need a `generate_mipmaps` property, so for now I just enabled it by default since that's what is done in 3.x.

Reply:
Ah, yes, I see the issue, in other cases we are only really checking if the immediate next canvas item uses mipmaps. So when doing the backbuffercopy we don't know whether mipmaps are needed at the time we do the copy. 

What we should do then is have another condition to check, something like:
```
if (material_screen_texture_cached && !mipmaps_generated && uses_mipmaps) {
    // generate mipmaps
}
```

So at the time of doing the copy, you just generate the base texture (unless you already know you need mipmaps), then once you need mipmaps you generate them. 

I really want to avoid always generating mipmaps. We did it in 3.x, but it introduced a lot of performance overhead and was an issue for a lot of projects.

Reply:
Ok, I implemented your suggestion and tested it.
However, there is an issue which needs to be addressed for this to be effective (maybe in a subsequent PR):
The ShaderCompiler seems to set uses_screen_texture_mipmaps whenever SCREEN_TEXTURE is used, not just when it's used together with textureLod (level > 0). Because of this, the mipmaps are always generated, even with the new logic. Not sure how to solve this elegantly.